packages/shadydom/src/patch-events.js

```diff
@@ -508,12 +518,12 @@ export function addEventListener(type, fnOrObj, optionsOrCapture) {
     wrapperFn: wrapperFn
   });
 
-  if (nonBubblingEventsToRetarget[type]) {
-    this.__handlers = this.__handlers || {};
-    this.__handlers[type] = this.__handlers[type] ||
-      {'capture': [], 'bubble': []};
-    this.__handlers[type][capture ? 'capture' : 'bubble'].push(wrapperFn);
-  } else {
+  this.__handlers = this.__handlers || {};
+  this.__handlers[type] = this.__handlers[type] ||
+    {'capture': [], 'bubble': []};
+  this.__handlers[type][capture ? 'capture' : 'bubble'].push(wrapperFn);
+
+  if (!nonBubblingEventsToRetarget[type]) {
```

Comment:
These changes cause _all_ event listeners to be manually added to `__handlers` so that `shadyDispatchEvent` can run them itself. Before, it was just things in `nonBubblingEventsToRetarget` (`focus`, `blur`).

Reply:
This was unexpected for me since I know we have to keep track of the handlers for `removeEventListener`. However, there we keep a map of wrapper function to node/options and here we need the reverse map from node to wrapper function.